grapher/stackedCharts/StackedDiscreteBarChart.tsx

```diff
@@ -362,6 +379,209 @@ export class StackedDiscreteBarChart
         )
     }
 
+    private renderBar(bar: Bar, tooltipProps: TooltipProps) {
+        const { axis, formatColumn, focusSeriesName, barHeight } = this
+        const { point, color, seriesName } = bar
+
+        const isFaint =
+            focusSeriesName !== undefined && focusSeriesName !== seriesName
+        const barX = axis.place(this.x0 + point.valueOffset)
+        const barWidth = axis.place(point.value) - axis.place(this.x0)
+
+        // Compute how many decimal places we should show.
+        // Basically, this makes us show 2 significant digits, or no decimal places if the number
+        // is big enough already.
+        const dp = Math.ceil(-Math.log10(point.value) + 1)
+        const barLabel = formatColumn.formatValueShort(point.value, {
+            numDecimalPlaces: isFinite(dp) && dp >= 0 ? dp : 0,
+        })
```

Comment:
Looks very similar to https://github.com/owid/owid-grapher/blob/9bc2033bbdc0e834d0d8589a080232d602e6f84e/grapher/axis/Axis.ts#L243-L245

Could some util be extracted?

Reply:
In theory, yes, and this is also where I took this from.
However, I wouldn't know what to name that? `detectFirstSignificantDigit` maybe, but then you would also need to clarify what a return value of 2 means?

Somewhat related, I was actually thinking of the [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat) API while working on this. They combine the options `minimumFractionDigits`, `minimumSignificantDigits`, `maximumFractionDigits`, and `maximumSignificantDigits` to achieve a very similar effect that we do in a different way. Maybe our number formatting should also work this way, or even use the `Intl.NumberFormat` API internally (using only the English preset)?